DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs

```diff
@@ -149,68 +151,131 @@ private async Task FetchConfigAsyncWithTask()
+                        var lastModified = res.ContentHeaders?.FirstOrDefault(e => e.Name?.ToLower() == "last-modified")
+                            ?.Value as string;
+                        var etag = res.Headers?.FirstOrDefault(e => e.Name?.ToLower() == "etag")?.Value as string;
+                        if (!string.IsNullOrEmpty(configLastModified) && lastModified != null &&
+                            !string.IsNullOrEmpty(lastModified))
                         {
-                            logger.LogWarning("Received timestamp on last-modified that was before the stored one. Not updating config.");
-                            return;
+                            var parsedHeader = Convert.ToDateTime(lastModified);
+                            var storedHeader = Convert.ToDateTime(configLastModified);
+                            // negative means that the stored header is before the returned parsed header
+                            if (DateTime.Compare(storedHeader, parsedHeader) >= 0)
+                            {
+                                logger.LogWarning(
+                                    "Received timestamp on last-modified that was before the stored one. Not updating config.");
+                                return;
+                            }
                         }
+
+                        try
+                        {
+                            var minimalConfig = JsonDocument.Parse(res.Content);
+                            var sseProp = minimalConfig.RootElement.GetProperty("sse");
+                            var sseUri = sseProp.GetProperty("hostname").GetString() +
+                                         sseProp.GetProperty("path").GetString();
+                            if (sseManager == null && localOptions.EnableBetaRealtimeUpdates)
+                            {
+                                sseManager = new SSEManager(sseUri, SSEStateHandler, SSEMessageHandler,
+                                    SSEErrorHandler);
+                                await sseManager.StartSSE();
+                            }
+                            else if (sseManager != null && localOptions.EnableBetaRealtimeUpdates)
+                            {
+                                sseManager.RestartSSE(sseUri);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            logger.LogWarning(e, "Failed to parse SSE config. Skipping SSE Initialization");
+                        }
+
+                        localBucketing.StoreConfig(sdkKey, res.Content);
+                        configEtag = etag;
+                        configLastModified = lastModified;
+                        logger.LogDebug("Config successfully initialized with etag: {ConfigEtag}, {lastmodified}",
+                            configEtag, configLastModified);
+                        //eventQueue?.QueueSDKConfigEvent(request, res);
```

Comment:
cleanup?

Reply:
yeah - was gonna stack sdkconfig on top of this